src/Microsoft.ApplicationInsights/Extensibility/W3C/W3CUtilities.cs

```diff
@@ -12,6 +13,27 @@
     public static class W3CUtilities
     {
         private static readonly uint[] Lookup32 = CreateLookup32();
+        private static readonly Regex TraceIdRegex = new Regex("^[a-f0-9]{32}$", RegexOptions.Compiled);
+
+        /// <summary>
+        /// Constructs a Telemetry ID from given traceid and span id in the format |traceid.spanid.
+        /// This is the format used by Application Insights.        
+        /// </summary>
+        /// <returns>constructed Telemetry ID.</returns>
+        public static string FormatTelemetryId(string traceId, string spanId)
+        {
+            return string.Concat("|", traceId, ".", spanId, ".");
+        }
+
+        /// <summary>
+        /// Checks if the given string is a valid trace-id as per W3C Specs.
+        /// https://github.com/w3c/distributed-tracing/blob/master/trace_context/HTTP_HEADER_FORMAT.md#trace-id .
+        /// </summary>
+        /// <returns>true if valid w3c trace id, otherwise false.</returns>
+        public static bool IsCompatibleW3CTraceID(string traceId)
```

Comment:
don't make it public. copy-paste is better. also, it should be more performant to check it without regex (each char in value range and length is 32)

Reply:
sticking with Regex now, as this is not a hot-path code. (only used in StartOperation(with operationid) which is not common). let me know if this is a common use case.

Reply:
this will be common in web/aspnet core, but there we can implement more efficient check.